### PR TITLE
Fix up and reinstate #1646 to not remove infrastructure units

### PR DESCRIPTION
**Pull request by simon33-2**
This seems to resolve the issue in Greyhawk_Wars from #1952 #1953.

Works in WW2 v5 & Classic so I'm not sure of any other cases (Global has the same rules as v5 for these purposes).

### CODE REVIEW

**Comment by DanVanAtta** [comment]
@simon33-2 can you walk us through the testing for this?

**Comment by simon33-2** [comment]
There's 4 cases I'm aware of.
- Attack at sea with only defenseless transports remaining
- Attack on land with destroyable AA Guns
- Attack on land with convertible AA Guns
- The Greyhawk Wars case described in #1952.

**Comment by ron-murhammer** [comment]
@simon33-2 So what about the following case:
- have a group of attacking units with total attack power > 0
- have a group of defending units (say 10) with total defense power = 0
- want to attack for 1 round and retreat so I can kill of some of them but don't want to take the territory due to potential counter attack

I believe that is possible to have a map like that today and this change would break that. Is that a valid case?

**Comment by simon33-2** [comment]
In that case you should not be able to retreat, at least in any of the Axis and Allies games.

That is the bug referred to in #1270.

**Comment by simon33-2** [comment]
Does it need a map property, if there are maps where it is valid to not take the territory?

**Comment by DanVanAtta** [comment]
@simon33-2 sorry to say, though this is incredibly hard to follow and nobody is shepherding this. I reviewed the description over the last 10 minutes, it's daunting to see text from: #1952 and that the updates in #1953 references more updates.

I do not wish for PR's to be sitting for upwards of a month!

I think more info is needed before this can really be considered in its due merit. At this point is where it becomes apparent that TripleA is a pain in the ass project. It seems simple because all the code is so straight forward and procedural, but then when it comes time to design and testing you start seeing all the technical debt and extra time you have to pay.

For this PR:
- what was broken exactly?
- are we adding something back in that was removed? If so, why was it removed? Is there any trade off we are making by adding these updates?
- can you go through all test scenarios and note the results. Again, here is where TripleA is a pain in the ass, it's up to you here @simon33-2  to do a ton of testing and to present the results here. It may be worth while to bite the bullet, extract this code to a micro class and test it there. Then the manual testing is cut down drastically at least.

**Comment by ron-murhammer** [comment]
@simon33-2 So I think this change should be alright now but can you verify that if it removes transports, does it properly remove any dependent land units as well?

**Comment by simon33-2** [comment]
Thanks @ron-murhammer 

@DanVanAtta . The problem with #1646 was that there was a case I wasn't aware of.
